### PR TITLE
Feature/staff assignments view

### LINKED ISSUE #1224: Staff assignment view to see staff current workload
- Account name or name of person: Di

**Is your feature request related to a problem? Please describe.**
We need to count how many reviews have been assigned to a specific person. Example: How many reviews Di has been assigned? What is the number of applications she is the primary reviewer and the number of applications she is the secondary reviewer?

**Is your feature request related to an existing functionality? Please describe.**
Feature might be related to creating tags to filter or sort applications into sub-categories

**Describe the solution you'd like**
We currently have filters for reviewers, but this filter does not distinguish primary from secondary. We require each person to sign up for about the same amount of primary and secondary to ensure equity and even distribution of applications. 

**Describe alternatives you've considered**
 #613 is a suggested feature to randomly assign reviewers. This new feature request might be similar, however, reviewers with expertise in a certain area are more likely to gravitate towards certain themes. A random assignment would not take this into consideration. 

**Priority**
- Low priority (annoying, would be nice to not see)

**Affected roles**
- Staff

**Ideal deadline**
Q4

### PR DESCRIPTION
**Pull request by frjo**
Fixes #1224

### CODE REVIEW

File: hypha/apply/funds/views.py [hunk 1]
```diff
@@ -1230,9 +1235,29 @@ class ReviewerLeaderboardDetail(SingleTableMixin, ListView):
...
+    template_name = 'funds/staff_assignments.html'
+
+    def get_queryset(self):
+        # Only list staff.
+        return self.model.objects.staff()
+
+    def get_table_data(self):
+        reviewer_roles = ReviewerRole.objects.all().order_by('order')
+        reviewer_role_id_1 = reviewer_roles.first().id
+        reviewer_role_id_2 = reviewer_roles.last().id
```
Comment (frjo) [inline hypha/apply/funds/views.py:1259]: @Parbhat I would like to dynamically create a column for each reviewer role. OTF happens to have two but the system allows you to create an arbitrary number of them. Any neat way to do this?

File: hypha/apply/funds/views.py [hunk 2]
```diff
@@ -1230,9 +1235,29 @@ class ReviewerLeaderboardDetail(SingleTableMixin, ListView):
...
+        return super().get_table_data().annotate(
+            role1=Count('assignedreviewers', filter=Q(assignedreviewers__role_id=reviewer_role_id_1)),
+            role2=Count('assignedreviewers', filter=Q(assignedreviewers__role_id=reviewer_role_id_2)),
+        )
```
Comment (frjo) [inline hypha/apply/funds/views.py:1263]: @Parbhat I would like to filter this down to only include assigned reviewers on submissions that are in status "active_statuses". Any tips?

Comment (frjo) [reply hypha/apply/funds/views.py:1263]: @Parbhat Here also I would like to accommodate an arbitrary number of reviewer roles if possible.

File: hypha/apply/funds/tables.py [hunk 3]
```diff
@@ -443,3 +443,18 @@ class Meta:
         order_by = ('-created_at',)
         attrs = {'class': 'all-reviews-table'}
         empty_text = _('No reviews available')
+
+
+class StaffAssignmentsTable(tables.Table):
+    full_name = tables.LinkColumn('funds:submissions:list', orderable=True, verbose_name="Staff", attrs={'td': {'class': 'title'}})
```
Comment (frjo) [inline hypha/apply/funds/tables.py:449]: @Parbhat I would like to add a query to this linked column so the view gets filtered on the reviewer.

`…/apply/submissions/all/?reviewers=123`

Comment (frjo) [reply hypha/apply/funds/tables.py:449]: or easier to just create a separate view for this?

File: hypha/apply/funds/views.py (hunk 1 above)
Comment (Parbhat) [reply hypha/apply/funds/views.py:1259]: `django-tables2` Table has an [extra_columns](https://django-tables2.readthedocs.io/en/latest/pages/api-reference.html#table) parameter to dynamically add columns. As we use `SimpleTableMixin`, these can be included in [get_table_kwargs](https://github.com/jieter/django-tables2/blob/6f5b5b6b228c3fc3a2491b83fe2c20f0e744dd01/django_tables2/views.py#L126) method.

File: hypha/apply/funds/views.py (hunk 2 above)
Comment (Parbhat) [reply hypha/apply/funds/views.py:1263]: Similar to the role, we can add a [submission](https://github.com/OpenTechFund/hypha/blob/32679916c2f9e5e82322296b14966659b5d708b9/hypha/apply/funds/models/submissions.py#L967) query to check status is in the active statuses.

Comment (Parbhat) [reply hypha/apply/funds/views.py:1263]: Looks like we do not need to annotate here as can use `extra_columns` parameter. More details in another comment.

File: hypha/apply/funds/tables.py [hunk 4]
```diff
@@ -443,3 +443,18 @@ class Meta:
...
+    reviewer_roles = ReviewerRole.objects.all().order_by('order')
+    role1 = tables.Column(verbose_name=reviewer_roles.first())
+    role2 = tables.Column(verbose_name=reviewer_roles.last())
```
Comment (Parbhat) [inline hypha/apply/funds/tables.py:452]: This code will run only when the code loads but will not update when data changes in db. We may not need this by using `extra_columns`. More details in another comment.

File: hypha/apply/funds/tables.py (hunk 3 above)
Comment (Parbhat) [reply hypha/apply/funds/tables.py:449]: For filtering, we can use `django-filters`'s FilterView like https://github.com/OpenTechFund/hypha/blob/32679916c2f9e5e82322296b14966659b5d708b9/hypha/apply/funds/views.py#L1009

**Comment by frjo** [comment]
@Parbhat Could your team take this on? The issue is about filtering on primary and secondary reviewer but I want to test this staff assignments table and see if that does not solve the need OTF staff have in a better way.

They want to get a picture of the workload put on staff. So the table should list all staff and what current submissions they are assigned to and it what reviewer role.

**Comment by Parbhat** [comment]
@frjo Yeah, will pick it.

**Comment by Parbhat** [comment]
@frjo PR updated with some updates. Can you check if it meets requirements?

**Comment by frjo** [comment]
@Parbhat Looks perfect, neat and easy to understand as all your code.

**Pull request merged**